tests/components/rest/test_binary_sensor.py

```diff
@@ -1,267 +1,381 @@
-                        "username": "my username",
-                        "password": "my password",
-                        "headers": {"Accept": CONTENT_TYPE_JSON},
-                    }
-                },
-            )
-            self.hass.block_till_done()
-        assert 1 == mock_req.call_count
-
-    @requests_mock.Mocker()
-    def test_setup_post(self, mock_req):
-        """Test setup with valid configuration."""
-        mock_req.post("http://localhost", status_code=200)
-        with assert_setup_component(1, "binary_sensor"):
-            assert setup_component(
-                self.hass,
-                "binary_sensor",
-                {
-                    "binary_sensor": {
-                        "platform": "rest",
-                        "resource": "http://localhost",
-                        "method": "POST",
-                        "value_template": "{{ value_json.key }}",
-                        "payload": '{ "device": "toaster"}',
-                        "name": "foo",
-                        "verify_ssl": "true",
-                        "authentication": "basic",
-                        "username": "my username",
-                        "password": "my password",
-                        "headers": {"Accept": CONTENT_TYPE_JSON},
-                    }
-                },
-            )
-            self.hass.block_till_done()
-        assert 1 == mock_req.call_count
-
-
-class TestRestBinarySensor(unittest.TestCase):
-    """Tests for REST binary sensor platform."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.rest = Mock("RestData")
-        self.rest.update = Mock(
-            "RestData.update", side_effect=self.update_side_effect('{ "key": false }')
-        )
-        self.name = "foo"
-        self.device_class = "light"
-        self.value_template = template.Template("{{ value_json.key }}", self.hass)
-        self.force_update = False
-        self.resource_template = None
-
-        self.binary_sensor = rest.RestBinarySensor(
-            self.hass,
-            self.rest,
-            self.name,
-            self.device_class,
-            self.value_template,
-            self.force_update,
-            self.resource_template,
-        )
-        self.addCleanup(self.hass.stop)
 
-    def update_side_effect(self, data):
-        """Side effect function for mocking RestData.update()."""
-        self.rest.data = data
 
-    def test_name(self):
-        """Test the name."""
-        assert self.name == self.binary_sensor.name
+async def test_setup_missing_basic_config(hass):
+    """Test setup with configuration missing required entries."""
+    assert await async_setup_component(
+        hass, binary_sensor.DOMAIN, {"binary_sensor": {"platform": "rest"}}
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)